pkg/kube/inject/inject.go

```diff
@@ -889,3 +898,61 @@ func GetProxyIDs(namespace *corev1.Namespace) (uid int64, gid int64) {
 
 	return
 }
+
+// GetProxyIDsFromPod returns the UID and GID to be used in the RunAsUser and RunAsGroup fields in the template
+// It relies on OpenShift injecting a valid UID into the pod template which we can then reuse.
+func GetProxyIDsFromPod(pod *corev1.Pod) (uid int64, gid int64) {
+	var proxyUID, proxyGID *int64
+	tproxyInterceptionMode := pod.Annotations != nil && pod.Annotations["sidecar.istio.io/interceptionMode"] == string(model.InterceptionTproxy)
+	if tproxyInterceptionMode {
+		proxyUID = ptr.To(int64(0))
+		proxyGID = ptr.To(constants.DefaultProxyUIDInt)
+	} else {
```

Comment:
note that I massively simplified the old function. Previously, we would look at the pod after injection, so we could distinguish between gateway and application cases based on number of containers and container names. This isn't really possible anymore as this now feeds into the normal injection (so we're looking at the pod before it is injected), but I think it's also not necessary.